src/addr.rs

```diff
@@ -546,6 +550,24 @@ impl PhysAddr {
     }
 }
 
+#[cfg(feature = "memory_encryption")]
+impl PhysAddr {
+    /// Creates a new physical address, throwing bits 52..64 and the encryption bit away.
+    #[inline]
+    pub fn new_truncate(addr: u64) -> PhysAddr {
+        PhysAddr((addr % (1 << 52)) & !ENC_BIT_MASK.load(Ordering::Relaxed))
+    }
+}
+
+#[cfg(not(feature = "memory_encryption"))]
+impl PhysAddr {
+    /// Creates a new physical address, throwing bits 52..64 away.
+    #[inline]
+    pub const fn new_truncate(addr: u64) -> PhysAddr {
+        PhysAddr(addr % (1 << 52))
+    }
+}
+
```

Comment:
I'd prefer it if we kept these in the original impl blocks next to the other functions and put the `#[cfg(...)]` selectors on the functions instead of the whole impl block.

Also applies to `PageTableEntry`.